backend/src/handlers/libraries_handler.go

```diff
@@ -0,0 +1,74 @@
+package handlers
+
+import (
+	"UnlockEdv2/src/models"
+	"net/http"
+	"strconv"
+)
+
+func (srv *Server) registerLibraryRoutes() {
+	srv.Mux.Handle("GET /api/libraries", srv.applyAdminMiddleware(srv.handleIndexLibraries))
+	srv.Mux.Handle("GET /api/libraries/visible", srv.applyMiddleware(srv.handleIndexVisibleLibraries))
+	srv.Mux.Handle("PUT /api/libraries/{id}", srv.applyAdminMiddleware(srv.handleToggleLibraryVisibility))
+}
+
+func (srv *Server) handleIndexLibraries(w http.ResponseWriter, r *http.Request, log sLog) error {
```

Comment:
Can we have one endpoint/handler that shows the libraries, and the request context will determine what is being shown?

```go
user := r.Context().Value("ClaimsKey").(*Claims)
if user.Role != models.Admin {
   // only show visible. and eventually it will look up their individual permissions
}
```